Ignore periods in Gmail addresses

mbrukman opened the pull request:
Fixes #23

mbrukman commented:
@DiSiqueira — another PR for your review, whenever you get a chance. Thanks in advance!

DiSiqueira commented:
👍 LGMT

But I would like to point out that there are at least two other ways you can create automatic aliases for Gmail accounts.
1) [using `@googlemail.com` instead of `gmail.com`](https://gmailgeek.com/facts/gmail-vs-google-mail-the-difference-explained.html)
2) [appending `+anything` before the domain.](https://www.eringibson.co/blog/how-to-create-unlimited-testing-accounts-with-the-gmail-hack)

```username@googlemail.com
username+crb@gmail.com
username+crb@googlemai.com```

Are also valid Gmail addresses that can be used for `username@gmail.com`.

mbrukman commented:
Thanks for the review, @DiSiqueira! You're absolutely right, `@googlemail.com` is also a valid Gmail domain as is `@gmail.com` and should be treated similarly with regard to placement of spaces. I've fixed that and updated the tests as well just now.

Regarding the `username+foo@gmail.com`: I think that's an explicit decision by a user to modify their email address, and while I agree (and you're correct) that the email will be delivered to the same email address, it's not as much of an "accident" as when adding or leaving periods, or capitalization, which is what I'm trying to protect against here.

In my past experience, when I've also used a similar trick, I meant to use the `username+foo@...` everywhere, to make sure that the email was filtered properly, or that I would know who managed to leak my email address, or where my email was found to contact me.

In other words, if someone is using the `+suffix`, they should use it consistently everywhere, from the CLA, to their git user config, and shouldn't rely on the CLA email canonicalizer to equate them (even though email systems do). Does that make sense?

DiSiqueira commented on ghutil/ghutil_test.go:
```diff
@@ -225,3 +225,91 @@ func TestMatchAccount_DoesNotMatchCase(t *testing.T) {
+	}
+}
+
+func TestGoogleMailAddress_PeriodsDoNotMatchCLA(t *testing.T) {
```
Can we extract the test and accept the `accountVsCla` as a parameter? It would avoid the code duplication here.

mbrukman replied:
I combined the two test cases since they're identical, other than the `@gmail.com` vs. `@googlemail.com` email addresses — does that work?

DiSiqueira commented:
👍 You have a good point, completely agree with it. The user should use the `+suffix` consistelly if he chooses to use it.
